### ISSUE TITLE
trouble of compile

### ISSUE THREAD

**Issue by hwang-happy** [Shark-ML/Shark#132]
**hi sharkers, i install shark-ML on my linux cluster, and everything is ok.

my gcc version is 5.10 and boost version is 1.62.0.

and i type  ./SharkVersion , and everything is ok.**
$ ./SharkVersion 
{
    "version": {
        "major": "3",
        "minor": "1",
        "patch": "0"
    },
    "isOfficialRelease": "false",
    "platform": "linux",
    "compiler": "GNU C++ version 5.1.0",
    "stdLib": "GNU libstdc++ version 20150422",
    "boostVersion": {
        "major": "1",
        "minor": "62",
        "patch": "0"
    },
    "buildType": "Release",
    "dynamicBuild": "false",
    "hasOpenMp": "true"
}

Then i write a example scripts like below:
**#include <shark/Data/Csv.h>                        
#include <shark/Algorithms/Trainers/CARTTrainer.h>  
#include <shark/ObjectiveFunctions/Loss/ZeroOneLoss.h>
//ing namespace shark;
using namespace std;
int main(){
    shark::ClassificationDataset dataTrain;
    importCSV(dataTrain, "data/C.csv", LAST_COLUMN, ' ');
   ClassificationDataset dataTest =splitAtElement(dataTrain,311);
   cout << "Training set - number of data points: " << dataTrain.numberOfElements()
         << " number of classes: " << numberOfClasses(dataTrain)
         << " input dimension: " << inputDimension(dataTrain) << endl;

   cout << "Test set - number of data points: " << dataTest.numberOfElements()
         << " number of classes: " << numberOfClasses(dataTest)
         << " input dimension: " << inputDimension(dataTest) << endl;

    std::cout<<"hello.shark"<<std::endl;
    return 0;
    };**

compile command 
**g++ ./sh.cpp -I /home/hwang/softwear/Programe/libc++/shark/include -I /home/hwang/softwear/Programe/libc++/boost1.62.0/include -L /home/hwang/softwear/Programe/libc++/boost1.62.0/lib -L /home/hwang/softwear/Programe/libc++/shark/lib64 -std=c++11 -o shark**

but errors return 

/tmp/cceKrERZ.o: In function `boost::serialization::void_downcast(boost::serialization::extended_type_info const&, boost::serialization::extended_type_info const&, void*)':
sh.cpp:(.text._ZN5boost13serialization13void_downcastERKNS0_18extended_type_infoES3_Pv[_ZN5boost13serialization13void_downcastERKNS0_18extended_type_infoES3_Pv]+0x27): undefined reference to `boost::serialization::void_downcast(boost::serialization::extended_type_info const&, boost::serialization::extended_type_info const&, void const*)'
/tmp/cceKrERZ.o: In function `void boost::serialization::throw_exception<boost::archive::archive_exception>(boost::archive::archive_exception const&)':
sh.cpp:(.text._ZN5boost13serialization15throw_exceptionINS_7archive17archive_exceptionEEEvRKT_[_ZN5boost13serialization15throw_exceptionINS_7archive17archive_exceptionEEEvRKT_]+0x25): undefined reference to `boost::archive::archive_exception::archive_exception(boost::archive::archive_exception const&)'
sh.cpp:(.text._ZN5boost13serialization15throw_exceptionINS_7archive17archive_exceptionEEEvRKT_[_ZN5boost13serialization15throw_exceptionINS_7archive17archive_exceptionEEEvRKT_]+0x2a): undefined reference to `boost::archive::archive_exception::~archive_exception()'
sh.cpp:(.text._ZN5boost13serialization15throw_exceptionINS_7archive17archive_exceptionEEEvRKT_[_ZN5boost13serialization15throw_exceptionINS_7archive17archive_exceptionEEEvRKT_]+0x2f): undefined reference to `typeinfo for boost::archive::archive_exception'
/tmp/cceKrERZ.o: In function `boost::archive::detail::iserializer<boost::archive::polymorphic_iarchive, shark::detail::SharedContainer<shark::blas::vector<double> > >::iserializer()':
sh.cpp:(.text._ZN5boost7archive6detail11iserializerINS0_20polymorphic_iarchiveEN5shark6detail15SharedContainerINS4_4blas6vectorIdEEEEEC2Ev[_ZN5boost7archive6detail11iserializerINS0_20polymorphic_iarchiveEN5shark6detail15SharedContainerINS4_4blas6vectorIdEEEEEC5Ev]+0x1f): undefined reference to `boost::archive::detail::basic_iserializer::basic_iserializer(boost::serialization::extended_type_info const&)'
/tmp/cceKrERZ.o: In function `boost::archive::detail::iserializer<boost::archive::polymorphic_iarchive, shark::detail::SharedContainer<shark::blas::vector<double> > >::~iserializer()':
sh.cpp:(.text._ZN5boost7archive6detail11iserializerINS0_20polymorphic_iarchiveEN5shark6detail15SharedContainerINS4_4blas6vectorIdEEEEED2Ev[_ZN5boost7archive6detail11iserializerINS0_20polymorphic_iarchiveEN5shark6detail15SharedContainerINS4_4blas6vectorIdEEEEED5Ev]+0x20): undefined reference to `boost::archive::detail::basic_iserializer::~basic_iserializer()'
/tmp/cceKrERZ.o: In function `boost::archive::detail::oserializer<boost::archive::polymorphic_oarchive, shark::detail::SharedContainer<shark::blas::vector<double> > >::oserializer()':

so is anything wrong with my Installation? how can i solve this problem? thanks for your help.

**Comment by hwang-happy** [comment]
#include <shark/Data/Csv.h>
#include <shark/Algorithms/Trainers/CARTTrainer.h>
#include <shark/ObjectiveFunctions/Loss/ZeroOneLoss.h>
using namespace shark;
using namespace std;
int main(){
ClassificationDataset dataTrain;
importCSV(dataTrain, "data/C.csv", LAST_COLUMN, ' ');
ClassificationDataset dataTest =splitAtElement(dataTrain,311);
cout << "Training set - number of data points: " << dataTrain.numberOfElements()
<< " number of classes: " << numberOfClasses(dataTrain)
<< " input dimension: " << inputDimension(dataTrain) << endl;

cout << "Test set - number of data points: " << dataTest.numberOfElements()
<< " number of classes: " << numberOfClasses(dataTest)
<< " input dimension: " << inputDimension(dataTest) << endl;

std::cout<<"hello.shark"<<std::endl;
return 0;
};**

**Comment by Ulfgard** [comment]
you miss to link to the required boost libraries and your linker is telling you exactly that.

http://image.diku.dk/shark/sphinx_pages/build/html/rest_sources/tutorials/first_steps/your_programs.html

________________________________
